src/game/SpellEffects.cpp

```diff
@@ -1149,6 +1149,13 @@ void Spell::EffectDummy(SpellEffectIndex eff_idx)
 
                     return;
                 }
+                case 32027:                                 // Expedition Flare
+                {
+                    if (!m_CastItem)                        // 32029 = Expedition Preserver | 32030 = Expedition Scout
```

Comment:
It depends on the actual logic of the effect.

If your target is an item, then we need to check the itemTarget.
If the target is a gameobject, we need to check for the gameobject.
Also, many effects have logic that will check for the TypeId() of caster or target.

But usually we don't need to check for m_caster, because that can't be NULL.
The caster was already used many times in the cast function before the spell effects are triggered.